network/acceptedwork.go

```diff
@@ -175,6 +176,10 @@ func ListMinedWork(db *bolt.DB) ([]*AcceptedWork, error) {
 
 			if work.Confirmed {
 				minedWork = append(minedWork, &work)
+
+				if truncate && len(minedWork) == truncatedSize {
```

Comment:
check length to avoid a potential unnecessary allocation?

```go
if truncate && (len(minedWork) + len(work)) == truncatedSize {
    return nil
}
minedWork = append(minedWork, &work)
```

Reply:
`work` would still need appending to `minedWork` after checking `if truncate && (len(minedWork) + len(work)) == truncatedSize` before returning, no?

Reply:
you are correct.